src/inmanta/deploy/scheduler.py

```diff
@@ -263,13 +264,26 @@ async def _new_version(
                 self._work.delete_resource(resource)
 
             new_desired_state: set[ResourceIdStr] = set()
+            # Only contains the direct undeployable resources, not the transitive ones.
+            blocked_resources: set[ResourceIdStr] = set()
+            # Resources that were undeployable in a previous model version, but got unblocked. Not the transitive ones.
+            unblocked_resources: set[ResourceIdStr] = set()
             added_requires: dict[ResourceIdStr, Set[ResourceIdStr]] = {}
             dropped_requires: dict[ResourceIdStr, Set[ResourceIdStr]] = {}
             for resource, details in resources.items():
-                if (
-                    resource not in self._state.resources
-                    or details.attribute_hash != self._state.resources[resource].attribute_hash
-                ):
+                if details.status is const.ResourceState.undefined:
+                    blocked_resources.add(resource)
+                    self._work.delete_resource(resource)
```

Comment:
This is not necessary, right? As we do this a bit below:
```
                for resource in blocked_resources | transitively_blocked_resources:
                    self._work.delete_resource(resource)
```

Reply:
I am not sure I understand what you mean. Here we delete the blocked resources (and the transitively blocked resources). This needs to happen under the scheduler lock, otherwise we risk these resources may get deployed. This is undesired. A bit below we delete the deleted resources. We do this outside the scheduler lock for improved performance, i.e. we don't want to keep the lock acquired longer than strictly required. If a deleted resource gets deployed one time too many, it's not a big deal.